policies/curity-phantom-token-auth-inbound/doc.md

```diff
@@ -0,0 +1,154 @@
+
+#### Exposing the runtime
+Depending on where the Curity Identity Server is deployed you might have to 
+expose the runtime node using a reverse proxy. One option is to use 
+[ngrok](https://curity.io/resources/learn/expose-local-curity-ngrok/) but 
+other solutions could also be used. 
+
+#### OAuth Tools
+With the server up and running and available you can use [OAuth Tools](https://oauth.tools/) to test 
+the configuration and make sure that you are able to obtain a token. I an opaque token is possible to
+obtain you are good to continue.
+
+### Set Environment Variables
+Before adding the policy, there are a few environment variables that will need
+to be set that will be used in the Curity Phantom Token Policy.
+
+1. In the [Zuplo Portal](https://portal.zuplo.com) open the **Environment
+   Variables** section in the <SettingsTabIcon /> **Settings** tab.
+
+2. Click **Add new Variable** and enter the name `INTROSPECTION_URL` in the name
+   field. Set the value to URL endpoint of the Curity Identity Server that handles introspection. Ex. 
+   `https://idsvr.example.com/oauth/v2/oauth-introspect`
+
+3. Click **Add new Variable** again and enter the name `CLIENT_ID` in the
+   name field. Set the value to ID of the client that you added the introspection capability to.
+
+4. Click **Add new Variable** again and enter the name `CLIENT_SECRET` in the
+   name field. Set the value to the secret of the client that you added the introspection capability to.
+   **Make sure to enable `is Secret?`.**
+   
+
+### Create the module
+In the [Zuplo Portal](https://portal.zuplo.com) open the **Route Designer**
+   in the <CodeEditorTabIcon /> **Files** tab then click `+` next to **Modules**, choose **Empty Module** to add a new module. 
+   Name the module `curity-phantom-token.ts`. Next paste the below code into the editor.
+
+```js
+// curity-phantom-token.ts
+import { ZuploRequest, ZuploContext, environment } from "@zuplo/runtime";
+
+export default async function (
+  request: ZuploRequest,
+  context: ZuploContext) {
+  const authHeader = request.headers.get("Authorization");
+
+  if (!authHeader) {
+    return new Response(`No Authorization header`, { status: 401 });
+  }
+
+  var token = getToken(authHeader);
+
+  if (!token) {
+    return new Response(`Failed to parse token from Authorization header`, { status: 401 });
+  }
+
+  function getJwt(): Promise<string> {
+    return (fetch(environment.INTROSPECTION_URL, {
+      headers: {
+        'Authorization': 'Basic ' + btoa(`${environment.CLIENT_ID}:${environment.CLIENT_SECRET}`),
+        'Accept': 'application/jwt',
+        'Content-Type': 'application/x-www-form-urlencoded'
+      },
+      method: 'POST',
+      body: 'token=' + token + '&token_type_hint=access_token',
+    }).then(function (response) {
+      return response.text();
+    }))
+    //TODO: Catch introspection error
+  };
+
+  context.waitUntil(getJwt());
+
+  var jwt = await (getJwt());
```

Comment:
Will the introspection endpoint be called only once? Currently, it seems as if it would be called twice. Could the `getJwt` method be an async one and just await the result of `fetch`?